### ISSUE TITLE
Install failure of perl-5.6.0 on NetBSD-1.4.2/i386

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2048]
Migrated from [rt.perl.org#3327](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3327) (status was 'resolved')

Searchable as RT3327$

**Comment by p5pRT** [comment]
### From giles@nemeton.com.au

#### Created by giles@nemeton.com.au



Perl 5\.6\.0 builds cleanly and passes all but the dubious stdio tests
on NetBSD\-1\.4\.2/i386\.  NetBSD\-1\.4\.2 is the most recent NetBSD release\.

'make install' fails\, not managing to load the Glob module\.

I tracked the cause of this to be that NetBSD\-1\.4\.2/i386 has a
dlopen\(\) that requires absolute pathnames\.  From the dlopen\(3\) manual
page&#8203;:

&nbsp;     "The argument can either be an absolute pathname or it
&nbsp;      can be of the form \`\`lib\<name>\.so\[\.xx\[\.yy\]\]'' \.\.\."

Adding a file&#8203;:

&nbsp;       perl\-5\.6\.0/ext/DynaLoader/hints/netbsd\.pl

as for OpenBSD containing&#8203;:

\# XXX Configure test needed?
\# Some NetBSDs seem to have a dlopen\(\) that won't accept relative paths
$self\->\{CCFLAGS\} = $Config\{ccflags\} \. ' \-DDLOPEN\_WONT\_DO\_RELATIVE\_PATHS';

and rebuilding perl allowed the install to run correctly\.

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=medium

Site configuration information for perl v5.6.0:

Configured by giles at Sun Jun  4 10:54:47 EST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=netbsd, osvers=1.4.2, archname=i386-netbsd
    uname='netbsd nemeton.com.au 1.4.2 netbsd 1.4.2 (nemeton) #1: wed may 17 21:42:46 est 2000 giles@nemeton.com.au:usrsrcsysarchi386compilenemeton i386 '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='cc', optimize='-O', gccversion=egcs-2.91.60 19981201 (egcs-1.1.1 release)
    cppflags='-fno-strict-aliasing'
    ccflags ='-fno-strict-aliasing'
    stdchar='char', d_stdstdio=undef, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='ld', ldflags =''
    libpth=/usr/lib
    libs=-lm -lc -lcrypt
    libc=/usr/lib/libc.so.12.40, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=define, ccdlflags=' '
    cccdlflags='-DPIC -fPIC ', lddlflags='-Bshareable '

Locally applied patches:
    


@INC for perl v5.6.0:
    /usr/local/perl-5.6.0/lib/5.6.0/i386-netbsd
    /usr/local/perl-5.6.0/lib/5.6.0
    /usr/local/perl-5.6.0/lib/site_perl/5.6.0/i386-netbsd
    /usr/local/perl-5.6.0/lib/site_perl/5.6.0
    /usr/local/perl-5.6.0/lib/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/giles
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/giles/bin:/usr/local/perl5.005_03/bin:/usr/local/sbin:/usr/local/bin:/usr/pkg/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/mh/bin:/usr/X11R6/bin:/var/qmail/bin:/usr/local/dnscache/bin:/usr/local/pgsql/bin:/usr/local/mysql/bin
    PERL_BADLANG (unset)
    SHELL=/bin/ksh


```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

#### Created by giles@nemeton.com.au



Perl 5\.6\.0 builds cleanly and passes all but the dubious stdio tests
on NetBSD\-1\.4\.2/i386\.  NetBSD\-1\.4\.2 is the most recent NetBSD release\.

'make install' fails\, not managing to load the Glob module\.

I tracked the cause of this to be that NetBSD\-1\.4\.2/i386 has a
dlopen\(\) that requires absolute pathnames\.  From the dlopen\(3\) manual
page&#8203;:

&nbsp;     "The argument can either be an absolute pathname or it
&nbsp;      can be of the form \`\`lib\<name>\.so\[\.xx\[\.yy\]\]'' \.\.\."

Adding a file&#8203;:

&nbsp;       perl\-5\.6\.0/ext/DynaLoader/hints/netbsd\.pl

as for OpenBSD containing&#8203;:

\# XXX Configure test needed?
\# Some NetBSDs seem to have a dlopen\(\) that won't accept relative paths
$self\->\{CCFLAGS\} = $Config\{ccflags\} \. ' \-DDLOPEN\_WONT\_DO\_RELATIVE\_PATHS';

and rebuilding perl allowed the install to run correctly\.

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=medium

Site configuration information for perl v5.6.0:

Configured by giles at Sun Jun  4 10:54:47 EST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=netbsd, osvers=1.4.2, archname=i386-netbsd
    uname='netbsd nemeton.com.au 1.4.2 netbsd 1.4.2 (nemeton) #1: wed may 17
21:42:46 est 2000 giles@nemeton.com.au:usrsrcsysarchi386compilenemeton i386 '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef
usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='cc', optimize='-O', gccversion=egcs-2.91.60 19981201 (egcs-1.1.1
release)
    cppflags='-fno-strict-aliasing'
    ccflags ='-fno-strict-aliasing'
    stdchar='char', d_stdstdio=undef, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t',
lseeksize=8
    alignbytes=4, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='ld', ldflags =''
    libpth=/usr/lib
    libs=-lm -lc -lcrypt
    libc=/usr/lib/libc.so.12.40, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=define, ccdlflags=' '
    cccdlflags='-DPIC -fPIC ', lddlflags='-Bshareable '

Locally applied patches:
    


@INC for perl v5.6.0:
    /usr/local/perl-5.6.0/lib/5.6.0/i386-netbsd
    /usr/local/perl-5.6.0/lib/5.6.0
    /usr/local/perl-5.6.0/lib/site_perl/5.6.0/i386-netbsd
    /usr/local/perl-5.6.0/lib/site_perl/5.6.0
    /usr/local/perl-5.6.0/lib/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/giles
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
   
PATH=/home/giles/bin:/usr/local/perl5.005_03/bin:/usr/local/sbin:/usr/local/bin:/usr/pkg/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/mh/bin:/usr/X11R6/bin:/var/qmail/bin:/usr/local/dnscache/bin:/usr/local/pgsql/bin:/usr/local/mysql/bin
    PERL_BADLANG (unset)
    SHELL=/bin/ksh


```
</details>

**Comment by p5pRT** [comment]
### From @smpeters

> \[RT\_System \- Wed Jun 07 00&#8203;:23&#8203;:43 2000\]&#8203;:
> 
> 
> This is a bug report for perl from giles@&#8203;nemeton\.com\.au\,
> generated with the help of perlbug 1\.28 running under perl v5\.6\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> Perl 5\.6\.0 builds cleanly and passes all but the dubious stdio tests
> on NetBSD\-1\.4\.2/i386\.  NetBSD\-1\.4\.2 is the most recent NetBSD release\.
> 
> 'make install' fails\, not managing to load the Glob module\.
> 
> I tracked the cause of this to be that NetBSD\-1\.4\.2/i386 has a
> dlopen\(\) that requires absolute pathnames\.  From the dlopen\(3\) manual
> page&#8203;:
> 
>      "The argument can either be an absolute pathname or it
>       can be of the form \`\`lib\<name>\.so\[\.xx\[\.yy\]\]'' \.\.\."
> 
> Adding a file&#8203;:
> 
>        perl\-5\.6\.0/ext/DynaLoader/hints/netbsd\.pl
> 
> as for OpenBSD containing&#8203;:
> 
> \# XXX Configure test needed?
> \# Some NetBSDs seem to have a dlopen\(\) that won't accept relative
>    paths
> $self\->\{CCFLAGS\} = $Config\{ccflags\} \. '
>    \-DDLOPEN\_WONT\_DO\_RELATIVE\_PATHS';
> 
> and rebuilding perl allowed the install to run correctly\.
> 

Patched with change \#6373\.

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
